plugins/filter_kubernetes/kube_meta.c

```diff
@@ -887,6 +887,11 @@ static int get_and_merge_meta(struct flb_kube *ctx, struct flb_kube_meta *meta,
                         meta->namespace, meta->podname,
                         &api_buf, &api_size);
 
+    /* ealy return if api server response is empty which prevents the result from being cached */
+    if (api_buf == NULL) {
```

Comment:
instead of checking ```api_buf == NULL``` which is a "valid" approach, the right approach is to catch the return value of ```get_api_server_info()```, and if It returned -1, just return -1, e.g:

```C
ret =  get_api_server_info(ctx,
                           meta->namespace, meta->podname,
                           &api_buf, &api_size);

if (ret == -1) {
    return -1;
}
```